includes/Linker.php

```diff
@@ -464,7 +464,11 @@ static function normaliseSpecialPage( Title $title ) {
 	 */
 	private static function fnamePart( $url ) {
 		if (VignetteRequest::isVignetteUrl($url)) {
-			return VignetteRequest::getImageFilename($url);
+			try {
+				return VignetteRequest::getImageFilename($url);
+			} catch (Exception $e) {
+				// VignetteUrlToUrlGenerator logs the exact error
```

Comment:
Yeah, makes sense.

I think it all depends on how we define "an exception" in this case. Is "broken" URL an edgecase here? Is this URL something we can modify (e.g. in the wikitext)? If so, then yes - moving `try/catch` block one level down sounds good to me

Sorry for being grumpy here, but I'm currently taking a deep look at what goes to our logs (fatals, exceptions and warnings) and we should definitely make them less noisy :)